### PR TITLE
NAS-107600 / 12.0 / Delete orphaned iSCSI shares when exporting pool

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/iscsi.py [hunk 1]
```diff
@@ -1588,15 +1588,20 @@ async def is_child_of_path(self, resource, path):
         return is_child(resource[self.path_field], os.path.join('zvol', os.path.relpath(path, '/mnt')))
 
     async def delete(self, attachments):
-        lun_ids = []
+        orphan_targets_ids = set()
         for attachment in attachments:
             for te in await self.middleware.call('iscsi.targetextent.query', [['extent', '=', attachment['id']]]):
+                orphan_targets_ids.add(te['target'])
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/iscsi.py:1594]: How can you know there are no more extents associated with this target?

Comment (themylogin) [reply src/middlewared/middlewared/plugins/iscsi.py:1594]: I discard the ones still associated with some extent later https://github.com/freenas/freenas/pull/5741/files#diff-a6a523dfb3e8a62b0d0efc1730d014c6R1601

Comment (william-gr) [reply src/middlewared/middlewared/plugins/iscsi.py:1594]: Ah thanks

**Review by william-gr — APPROVED** [review]

**Comment by themylogin** [comment]
backport

**Pull request merged**